src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -675,8 +682,12 @@ private void startCleaner(long recordOffset, SnowflakeTelemetryPipeCreation pipe
             }
           });
 
-      if (reprocessFiles.size() > 0) {
+
+      if (!cleanFilesOnlyIfStatusLoaded && reprocessFiles.size() > 0) {
```

Comment:
+1 
let's also disable `filterFileReprocess` above if reprocess-files' cleanup is disabled.